### PR TITLE
Use yarn 2

### PR DESCRIPTION
**Pull request by fisker**
- [ ] I’ve added tests to confirm my change works.
- [ ] (If changing the API or CLI) I’ve documented the changes I’ve made (in the `docs/` directory)
- [ ] (If the change is user-facing) I’ve added my changes to `changelog_unreleased/*/pr-XXXX.md` file following `changelog_unreleased/TEMPLATE.md`.
- [x] I’ve read the [contributing guidelines](https://github.com/prettier/prettier/blob/master/CONTRIBUTING.md).

**✨[Try the playground for this PR](https://prettier.io/playground-redirect)✨**

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -108,7 +108,7 @@
     "eslint-formatter-friendly": "7.0.0",
     "eslint-plugin-import": "2.22.1",
     "eslint-plugin-jest": "24.1.0",
-    "eslint-plugin-prettier-internal-rules": "file:scripts/tools/eslint-plugin-prettier-internal-rules",
+    "eslint-plugin-prettier-internal-rules": "file:scripts/tools/eslint-plugin-prettier-internal-rules/eslint-plugin-prettier-internal-rules-1.0.0.tgz",
```
Comment (fisker) [inline package.json:111]: Somehow, yarn calculate different checksum on windows and other os.

File: package.json [hunk 2]
```diff
@@ -23,14 +23,15 @@
...
+    "@fisker/parse-srcset": "1.0.2",
     "@glimmer/syntax": "0.62.4",
     "@iarna/toml": "2.2.5",
     "@typescript-eslint/typescript-estree": "3.10.1",
     "angular-estree-parser": "2.2.1",
     "angular-html-parser": "1.7.1",
     "camelcase": "6.2.0",
     "chalk": "4.1.0",
-    "ci-info": "watson/ci-info#f43f6a1cefff47fb361c88cf4b943fdbcaafe540",
+    "ci-info": "2.0.0",
```
Comment (fisker) [inline package.json:34]: Don't know why, but there two can't install in netlify

File: package.json (hunk 1 above)
Comment (fisker) [reply package.json]: Use `link:` instead of `file:` resolved this problem.

**Comment by thorn0** [comment]
What are the benefits of doing this change?

**Comment by fisker** [comment]
Not really, but yarn suggest to upgrade https://yarnpkg.com/getting-started/qa#why-should-you-upgrade-to-yarn-modern

**Comment by fisker** [comment]
Close in favor of #12599

**Pull request closed without merging**